inventory.ts

```diff
@@ -0,0 +1,222 @@
+
+        if (this.value === other.value) return this.name.toLowerCase() < other.name.toLowerCase() ? 1 : -1;
+
+        return this.value > other.value ? 1 : -1;
+
+    }
+
+    public toString(): string {
+        return `${this.name} − Value: ${this.value}, Weight: ${this.weight.toFixed(2)}`;
+    }
+}
+
+abstract class Consumable extends Item {
+    isConsumed: boolean;
+    isSpoiled: boolean;
+
+    constructor(name: string, value: number, weight: number, isSpoiled: boolean) {
+        super(name, value, weight)
+        this.isSpoiled = isSpoiled;
+        this.isConsumed = false;
+    }
+
+    eat(): string {
+        if (this.isConsumed) return `There is nothing left of the ${this.name} to consume.`;
+
+        this.isConsumed = true;
+        return `You eat the ${this.name}.`;
+    }
+
+    use(): string {
+        const spoiledText = this.isSpoiled ? ' You feel sick.' : '';
+
+        return `${this.eat()}${spoiledText}`;
+
+    }
+}
+
+abstract class Weapon extends Item {
+    baseDamage: number;
+    damageModifier: number;
+    baseDurability: number;
+    durabilityModifier: number;
+    isBroken: boolean;
+    MODIFIER_CHANGE_RATE: number;
+
+    constructor(name: string,
+        value: number,
+        weight: number,
+        baseDamage: number,
+        damageModifier: number,
+        baseDurability: number,
+        durabilityModifier: number
+    ) {
+        super(name, value, weight);
+
+        this.baseDamage = baseDamage;
+        this.damageModifier = damageModifier;
+        this.baseDurability = baseDurability;
+        this.durabilityModifier = durabilityModifier;
+
+        this.isBroken = false;
+        this.MODIFIER_CHANGE_RATE = 0.1;
+    }
+
+    public getDamage(): number {
+        return this.baseDamage + this.damageModifier;
+    };
+
+    public getDurability(): number {
+        return this.baseDurability + this.durabilityModifier;
+    };
+
+    public toString(): string {
+        const damage = this.getDamage().toFixed(2);
+        const durability = this.getDurability().toFixed(2);
+
+        return `${this.name} − Value: ${this.value}, Weight: ${this.weight}, Damage: ${damage}, Durability: ${durability}%`
+    }
+
+    public use(): string {
+        if (this.isBroken) return `You can't use the ${this.name}, it is broken.`
+
+        this.baseDurability = this.baseDurability - this.MODIFIER_CHANGE_RATE;
+
+        if (this.baseDurability <= 0) this.isBroken = true;
+        const damage = this.getDamage().toFixed(2);
+        const brokenText = this.isBroken ? ` The ${this.name} breaks.` : '';
+
+        return `You use the ${this.name}, dealing ${damage} points of damage.${brokenText}`;
+    }
+}
+
+class ItemWeightComparator implements ItemComparator {
+    compare(first: Item, second: Item) {
+        if (!first || !second) throw new Error('Items required!')
+
+        if (first.weight === second.weight) return first.compareTo(second)
+
+        return first.weight > second.weight ? 1 : -1;
```

Comment:
Too often construction a > b ? 1: -1;
Let's move it to function and use in all places.